### PR TITLE
feat(ui-kit): add new reusable components

### PR DESCRIPTION
**Pull request by iraboldyrieva**
## Describe your changes

This pull request introduces several new reusable components to the UI Kit:

- **DriverAvatar**: A component that displays a driver's initials inside a circular avatar.
- **DriverInfoCard**: Shows detailed information about a driver, including avatar, name, working hours, stops count, distance, and action icons.
- **OrderRow**: A row component displaying order details such as checkbox, date, time, address, luggage sizes, and driver information.
- **StatusBadge**: Displays status labels (Completed, Failed, Not Arrived, At Risk, Upcoming) with i18n support.
- **DetailedRouteRow**: Shows route details with city name, time, status badge, and action icons.
- **PopupMessage**: A customizable popup dialog with heading, messages, and action buttons.
- **BasicDatePicker** and **CalendarRange**: Date picker components using MUI's DatePicker and DateRangePicker with custom configurations.

## Issue ticket code (and/or) and link

- [SCRUM-1: UI Kit](https://codecraftersintership.atlassian.net/browse/SCRUM-1)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [ ] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [ ] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [ ] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [] No inline styles
- [x] Imports are absolute
- [ ] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/components/BasicDatePicker/index.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import * as React from 'react';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+
+const BasicDatePicker: React.FC = (props) => {
+  return (
+    <LocalizationProvider dateAdapter={AdapterDayjs}>
+      <DatePicker label="MM/DD/YYYY" {...props} />
```
Comment (lyudad) [inline src/components/BasicDatePicker/index.tsx]: 1.3.3

File: src/components/DetailedRouteRow/index.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import LocationOnIcon from '@mui/icons-material/LocationOn';
+import DeleteIcon from '@mui/icons-material/Delete';
+import Status from '../Status';
+import { COLORS } from '@/constants/colors';
+import { FONT } from '@/constants/font';
+
+interface DetailedRouteRowProps {
+  city: string;
+  time: string;
+  status: 'completed' | 'failed' | 'not_arrived' | 'at_risk' | 'upcoming';
```
Comment (lyudad) [inline src/components/DetailedRouteRow/index.tsx]: move to enum and to constants, I think we will reuse this enum in multiple places later

File: src/components/DriverInfoCard/index.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
+import { COLORS } from '@/constants/colors';
+import EditIcon from '@mui/icons-material/Edit';
+import MoreVertIcon from '@mui/icons-material/MoreVert';
+import VisibilityIcon from '@mui/icons-material/Visibility';
+import { Box, IconButton, Typography } from '@mui/material';
+import React from 'react';
```
Comment (lyudad) [inline src/components/DriverInfoCard/index.tsx]: я уже писала что реакт должен идти первым импортом

File: src/components/DriverInfoCard/index.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+import DriverAvatar from '../DriverAvatar';
+import { FONT } from '@/constants/font';
```
Comment (lyudad) [inline src/components/DriverInfoCard/index.tsx]: 1.1.14

File: src/components/PopupMessage/index.tsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import {
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Typography,
+} from '@mui/material';
+import { useTranslation } from 'react-i18next';
```
Comment (lyudad) [inline src/components/PopupMessage/index.tsx]: 1.1.14

File: src/components/RoutesRow/index.tsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+interface RoutesRowProps {
+  routeId: string;
+  date: string;
+  driverFirstName: string;
+  driverLastName: string;
+  driverPhone: string;
+  stopsCount: number;
+  workingHours: string;
+  distance: number;
+  status: 'completed' | 'failed' | 'not_arrived' | 'at_risk' | 'upcoming';
```
Comment (lyudad) [inline src/components/RoutesRow/index.tsx]: is something repeat its a red flag that it should not be in the component, move to to constant

File: src/components/Status/index.tsx [hunk 7]
```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { Box, Typography } from '@mui/material';
+import { useTranslation } from 'react-i18next';
+import { COLORS } from '@/constants/colors';
+import { FONT } from '@/constants/font';
+
+interface StatusProps {
+  status: 'completed' | 'failed' | 'not_arrived' | 'at_risk' | 'upcoming';
```
Comment (lyudad) [inline src/components/Status/index.tsx]: same

File: src/components/Status/index.tsx [hunk 8]
```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { Box, Typography } from '@mui/material';
+import { useTranslation } from 'react-i18next';
```
Comment (lyudad) [inline src/components/Status/index.tsx:2]: 1.1.14

File: src/components/TopRow/index.tsx [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+interface CategoryProps {
+  icon: React.ReactNode;
+  label: string;
+}
+
+const CategoryRow: React.FC = () => {
+  const { t } = useTranslation();
+
+  const categories: CategoryProps[] = [
```
Comment (lyudad) [inline src/components/TopRow/index.tsx]: it will reinitialise with each rerender, memoize it

File: src/components/BasicDatePicker/index.tsx (hunk 1 above)
Comment (iraboldyrieva) [reply src/components/BasicDatePicker/index.tsx]: Got it

File: src/components/DriverInfoCard/index.tsx (hunk 3 above)
Comment (iraboldyrieva) [reply src/components/DriverInfoCard/index.tsx]: My bad, will fix it

File: src/components/DetailedRouteRow/index.tsx (hunk 2 above)
Comment (iraboldyrieva) [reply src/components/DetailedRouteRow/index.tsx]: Got it

File: src/components/BasicDatePicker/index.tsx [hunk 10]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DatePicker, DatePickerProps } from '@mui/x-date-pickers/DatePicker';
+import { useTranslation } from 'react-i18next';
+import { Dayjs } from 'dayjs';
+
+const BasicDatePicker: React.FC<DatePickerProps<Dayjs>> = (props) => {
+  const { t } = useTranslation();
+  const dateFormat = t('MM/DD/YYYY');
```
Comment (lyudad) [inline src/components/BasicDatePicker/index.tsx]: зачем выводить в константу, сразу перевод при рендере, сложно читать код где много констант

Comment (lyudad) [reply src/components/BasicDatePicker/index.tsx]: и название переменной не верно, должно быть что то типа t(dataFormat)

File: src/components/DetailedRouteRow/index.tsx [hunk 11]
```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import NoteIcon from '@mui/icons-material/Note';
+import LocationOnIcon from '@mui/icons-material/LocationOn';
+import DeleteIcon from '@mui/icons-material/Delete';
+import Status from '../Status';
```
Comment (lyudad) [inline src/components/DetailedRouteRow/index.tsx]: 1.1.14

File: src/components/DetailedRouteRow/index.tsx [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+import { COLORS } from '@/constants/colors';
+import { FONT } from '@/constants/font';
+import { Box, Typography, IconButton } from '@mui/material';
+import { StatusEnum } from '@/constants/status';
```
Comment (lyudad) [inline src/components/DetailedRouteRow/index.tsx:5]: contans to constant., styles to styles, dont mess up

File: src/components/DriverInfoCard/index.tsx (hunk 4 above)
Comment (lyudad) [reply src/components/DriverInfoCard/index.tsx]: nothing changed, check 1.1.14

File: src/components/OrderRow/index.tsx [hunk 13]
```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import CustomCheckbox from '../Checkbox';
+import DriverAvatar from '../DriverAvatar';
+import { COLORS } from '@/constants/colors';
+import { FONT } from '@/constants/font';
+import { Box, Typography } from '@mui/material';
```
Comment (lyudad) [inline src/components/OrderRow/index.tsx:5]: та йомайо, настрой притир правила 1.1.14

File: src/components/PopupMessage/index.tsx (hunk 5 above)
Comment (lyudad) [reply src/components/PopupMessage/index.tsx]: same issue

**Review by lyudad — APPROVED** [review]

**Pull request merged**